### ISSUE TITLE
Potential Boot Loop

### ISSUE THREAD

**Issue by Wh1t3Fox** [Resinchem/ESP-Parking-Assistant#3]
I flashed the firmware on an ESP8266  D1 mini and an ESP8266 NodeMCU. I was able to access the web UI, but after entering my wifi information and LED numbers the device never allows me to access the web UI again. Nmap showed port 80 open and when I power cycle the device the flashing blue light never stops. Below is the output from the device:
```
[08:03:00] ets Jan  8 2013,rst cause:1, boot mode:(3,0)
[08:03:00]
[08:03:00]load 0x4010f000, len 3584, room 16 
[08:03:00]tail 0
[08:03:00]chksum 0xb0
[08:03:00]csum 0xb0
[08:03:00]v2843a5ac
[08:03:00]~ld
[08:03:00]
[08:03:00] ets Jan  8 2013,rst cause:1, boot mode:(3,0)
[08:03:00]
[08:03:00]load 0x4010f000, len 3584, room 16 
[08:03:00]tail 0
[08:03:00]chksum 0xb0
[08:03:00]csum 0xb0
[08:03:00]v2843a5ac
[08:03:00]~ld
[08:03:00]
[08:03:00] ets Jan  8 2013,rst cause:1, boot mode:(3,0)
[08:03:00]
[08:03:00]load 0x4010f000, len 3584, room 16 
[08:03:00]tail 0
[08:03:00]chksum 0xb0
[08:03:00]csum 0xb0
[08:03:00]v2843a5ac
[08:03:00]~ld
[08:03:01]
[08:03:01] ets Jan  8 2013,rst cause:1, boot mode:(3,0)
[08:03:01]
[08:03:01]load 0x4010f000, len 3584, room 16 
[08:03:01]tail 0
[08:03:01]chksum 0xb0
[08:03:01]csum 0xb0
[08:03:01]v2843a5ac
[08:03:01]~ld
[08:03:01]
[08:03:01] ets Jan  8 2013,rst cause:1, boot mode:(3,0)
[08:03:01]
[08:03:01]load 0x4010f000, len 3584, room 16 
[08:03:01]tail 0
[08:03:01]chksum 0xb0
[08:03:01]csum 0xb0
[08:03:01]v2843a5ac
[08:03:01]~ld
```

**Comment by Resinchem** [comment]
Could you possibly send me a screenshot or list of the values that you have entered in the web interface/configuration right before this problem begins?  I may not be error trapping for certain combination of values.

Also, if possible, can you look on the back of your TFMini and see which version of the sensor you have?  There should be something on the board that says TFMini v1.x.

**Comment by Wh1t3Fox** [comment]
SSID: syzygy-iot
Password: 32 characters consisting of upper and lower case letters and numbers (originally I had a password with special characters and removed those)
# LEDS: 30

Will the TFMini need to be plugged in for the firmware to properly load? I haven't fully looked through the source code yet, but as of right now I was just attempting to boot the esp8266 without the LEDs or TFMini attached.

**Comment by Resinchem** [comment]
Yes... the TFMini needs to be attached.  I think I mention this in both the wiki and the blog.  The code attempts to establish a serial connection to the TFMini as part of the setup process.  Actually, I'm surprised you were able to get the hotspot and web interface to show without the TFMini.

Try it with the TFMini attached.  If you are still seeing the boot loop, I'd like to see a screenshot of **all** the values you entered on the main settings page of the web interface.  Something like an invalid value in the distances could be causing something like a divide by zero error and resulting the boot loop.  You need to enter all distance values as well, but the web app doesn't error check for that (but it probably should).

**Comment by Wh1t3Fox** [comment]
Ahh okay my bad. I'll give that a try

**Comment by Wh1t3Fox** [comment]
I'm still having the same issues. Everything is plugged into the D1 mini.  TFmini version is 1.8.1.

Here is the wifi settings I applied. I have not been able to get to the main settings page even though I can see the device on my network.

![Screenshot_20221130-100617.png](https://user-images.githubusercontent.com/1253352/204834841-bf99b7df-e70e-432f-96b8-cb7bd83f0440.png)
